.github/workflows/pkgci_test_amd_mi300.yml

```diff
@@ -19,7 +19,7 @@ on:
 
 jobs:
   test_mi300:
-    runs-on: nodai-amdgpu-mi300-x86-64
+    runs-on: linux-mi300-gpu-1
```

Comment:
https://github.com/iree-org/iree/actions/runs/12399057884/job/34613576470?pr=19517#step:8:195
```
 iree/runtime/src/iree/hal/drivers/hip/dynamic_symbols.c:160: UNAVAILABLE; HIP runtime library 'amdhip64.dll'/'libamdhip64.so' not available: please ensure installed and in dynamic library search path: 
  Tried: libamdhip64.so
    iree/runtime/src/iree/base/internal/dynamic_library_posix.c:165: NOT_FOUND; failed to load dynamic library (possibly not found on any search path): libamdhip64.so: cannot open shared object file: No such file or directory; creating driver for device 'hip'; resolving dependencies for 'module'
```

What drivers and software are installed on these new runners? Should we run under Docker (either in the runners themselves or in the job)?

Reply:
Yeah discussing with Jodie offline. We can use the rocm gh runner docker for the runner instantiation, but I think it is probably best to specify the docker in the job (workflow file), so it is visible to everyone what rocm is being used

Reply:
I think it may depend on how long it takes to download and start the docker image. If it takes 5 minutes to download a 70GB docker image, configuring that in the runner itself could help hide some latency from workflows?

We could revive these dockerfiles as needed:
* https://github.com/iree-org/base-docker-images/blob/main/dockerfiles/amdgpu_ubuntu_jammy_x86_64.Dockerfile
* https://github.com/iree-org/base-docker-images/blob/main/dockerfiles/amdgpu_ubuntu_jammy_ghr_x86_64.Dockerfile

If we need to use Docker, I'd definitely prefer to use _either_ upstream public images if no extra deps are needed _or_ those iree-org ones.

Reply:
That would be great. These jobs do need CMake and ninja but otherwise the deps are pretty minimal.

Reply:
Agreed that it's clearer to set a container image in the workflow, but I'm not sure about the image download time. Updated and let's see how long it takes.